source/fe/fe_values.cc

```diff
@@ -3665,17 +3665,20 @@ FEValues<dim,spacedim>::initialize (const UpdateFlags update_flags)
                                    *this->mapping,
                                    quadrature,
                                    this->finite_element_output);
-  Threads::Task<typename Mapping<dim,spacedim>::InternalDataBase *>
-  mapping_get_data = Threads::new_task (&Mapping<dim,spacedim>::get_data,
-                                        *this->mapping,
-                                        flags,
-                                        quadrature);
+  if (flags & update_mapping)
+    {
+      Threads::Task<typename Mapping<dim,spacedim>::InternalDataBase *>
+      mapping_get_data = Threads::new_task (&Mapping<dim,spacedim>::get_data,
+                                            *this->mapping,
+                                            flags,
+                                            quadrature);
+      this->mapping_data.reset (mapping_get_data.return_value());
```

Comment:
This is not useful because it immediately requires the value of the task you just created. You'll have to declare the task object outside the `if` and do the reset at the bottom of the function where it currently is (also guarded by an `if`).